doc/nrf/ug_fw_update.rst

```diff
@@ -0,0 +1,311 @@
+
+* :ref:`ug_fw_update_keys_openssl`
+* :ref:`ug_fw_update_keys_imgtool`
+* :ref:`ug_fw_update_keys_python`
+
+.. _ug_fw_update_keys_openssl:
+
+OpenSSL
+-------
+
+`OpenSSL`_ is installed by default with some packages and programs, like *git*.
+It supports many different types of keys, but not all of them are supported by |NSIB| and MCUboot.
+
+See below for examples of using OpenSSL to create some commonly used key types:
+
+.. tabs::
+
+   .. group-tab:: ECDSA
+
+      .. code-block:: console
+
+         openssl ecparam -name prime256v1 -genkey -noout -out priv.pem
+
+   .. group-tab:: RSA-2048
+
+      .. code-block:: console
+
+         openssl genrsa -out priv.pem 2048
+
+.. note::
+
+   ``priv.pem``, ``priv_*.pem``, and ``pub_*.pem`` are keys named arbitrarily and used as an example in this documentation.
+   You can name private and public keys used with Zephyr and the |NCS| as you prefer, as long as the files are in the ``.pem`` format.
+
+To see a complete list of the supported key types, call ``openssl help`` and ``openssl <key type> -help`` from a terminal.
+
+.. _ug_fw_update_keys_imgtool:
+
+Imgtool
+-------
+
+:ref:`mcuboot:imgtool` is a Python tool maintained by MCUboot.
+It provides functionalities for handling public/private key pairs.
+
+It is also available as a PyPI package that you can install via ``pip``.
+However, when working within the |NCS| framework, it is recommended to use the script that is included with the fork of MCUboot used by the |NCS|.
+
+See below for examples of imgtool used to create some commonly used key types:
+
+.. tabs::
+
+   .. group-tab:: ECDSA
+
+      .. code-block:: console
+
+         python3 bootlooader/mcuboot/scripts/imgtool.py keygen -t ecdsa-p256 -k priv.pem
+
+   .. group-tab:: RSA-2048
+
+      .. code-block:: console
+
+         python3 bootlooader/mcuboot/scripts/imgtool.py keygen -t rsa-2048 -k priv.pem
+
+For a full list of the supported types, use the ``--help`` argument with the tool or any of its commands.
+
+.. _ug_fw_update_keys_python:
+
+Python
+------
+
+The |NCS| ships with an internal script used for interacting with private/public keys with the |NSIB|.
+This script is maintained by the |NCS| team and utilizes the ``ecdsa`` package available from PyPI.
+As such, it is only valid for use with ECDSA keys.
+
+.. code-block:: console
+
+   python3 nrf/scripts/bootloader/keygen.py --private -o priv.pem
+
+.. _ug_fw_update_development_keys:
+
+Development keys
+================
+
+When testing the bootloader chain, generating and using custom signing keys is optional.
+If you do not provide your own keys through Kconfig options, the build system will automatically create debug keys, depending on the bootloader(s) compiled into the application.
+
+.. caution::
+
+   Keys that are automatically used or generated by bootloaders for image signature validation are intended for *development and/or debug use only*.
```

Comment:
`and/or` is not allowed. Rephrase to `only for development or debug use, or both`.